src/group/tests/kat_messages.rs

```diff
@@ -0,0 +1,512 @@
+
+    // We can't easily get a "natural" GroupInfo, so we just create one here.
+    let group_info = GroupInfo::new(
+        group_id.clone(),
+        GroupEpoch(0),
+        get_random_vec(ciphersuite.hash_length()),
+        get_random_vec(ciphersuite.hash_length()),
+        vec![Box::new(RatchetTreeExtension::new(ratchet_tree.clone()))],
+        ConfirmationTag(Mac {
+            mac_value: get_random_vec(ciphersuite.hash_length()),
+        }),
+        LeafIndex::from(0u32),
+    );
+    let group_secrets = GroupSecrets::random_encoded(ciphersuite, ProtocolVersion::default());
+    let public_group_state = group.export_public_group_state(&credential_bundle).unwrap();
+
+    // Create some proposals
+    let key_package_bundle =
+        KeyPackageBundle::new(&[ciphersuite_name], &credential_bundle, Vec::new()).unwrap();
+    let key_package = key_package_bundle.key_package();
+
+    let add_proposal = AddProposal {
+        key_package: key_package.clone(),
+    };
+    let update_proposal = UpdateProposal {
+        key_package: key_package.clone(),
+    };
+    let remove_proposal = RemoveProposal {
+        removed: random_u32(),
+    };
+
+    let psk_id = PreSharedKeyId::new(
+        PskType::External,
+        Psk::External(ExternalPsk::new(get_random_vec(ciphersuite.hash_length()))),
+        get_random_vec(ciphersuite.hash_length()),
+    );
+
+    let psk_proposal = PreSharedKeyProposal { psk: psk_id };
+    let reinit_proposal = ReInitProposal {
+        group_id,
+        version: ProtocolVersion::Mls10,
+        ciphersuite: ciphersuite_name,
+        extensions: vec![Box::new(RatchetTreeExtension::new(ratchet_tree.clone()))],
+    };
+    // We don't support external init proposals yet.
+    let external_init_proposal = vec![0u8];
+    // We don't support app ack proposals yet.
+    let app_ack_proposal = vec![0u8];
+    let joiner_key_package_bundle =
+        KeyPackageBundle::new(&[ciphersuite_name], &credential_bundle, Vec::new()).unwrap();
+
+    let add_proposal_pt = group
+        .create_add_proposal(
+            b"aad",
+            &credential_bundle,
+            joiner_key_package_bundle.key_package().clone(),
+        )
+        .unwrap();
+    let (commit_pt, welcome_option, _option_kpb) = group
+        .create_commit(
+            b"aad",
+            &credential_bundle,
+            &[&add_proposal_pt],
+            &[],
+            true,
+            None,
+        )
+        .unwrap();
+    let commit = if let MlsPlaintextContentType::Commit(commit) = commit_pt.content() {
+        commit.clone()
+    } else {
+        panic!("Wrong content of MLS plaintext");
+    };
+    let welcome = welcome_option.unwrap();
+    let mls_ciphertext_application = group
+        .create_application_message(b"aad", b"msg", &credential_bundle, random_u8() as usize)
+        .unwrap();
+    let mls_plaintext_application = group.decrypt(&mls_ciphertext_application).unwrap();
+
+    let mls_ciphertext = group
```

Comment:
Should we randomly pick which plaintext to encrypt? Or encrypt all of them? If it increases coverage, we should probably encrypt all of them.

Reply:
As far as I can see, there's only one field for MLSCiphertext that I can populate. Also, since the ciphertext is just decoded and encoded (not decrypted), the coverage should remain mostly the same, as the only thing that really changes is the content type. I've randomized the encryption target, which should cover all the encoding and decoding of any content type.